### PR TITLE
packaging-improvements

### PR DESCRIPTION
**Pull request by bpepple**
Some things that might be worth adding to the project.

1. Add tests to the sdist, so that anyone wishing to package it has the tests to verify the code. Note it might be useful to either use fixture or small sqlite db in the tests directories so in the future if anything changes at Comic Vine the test don't start failing for other users.

2. Add python-coverage & flake to the dev dependencies to help with code quality.  In particular, there are a lot of unnecessary imports that aren't needed that flake8 are warning about.

### CODE REVIEW

**Comment by Buried-In-Code** [comment]
Hi, thanks so much for your help. There's clearly still a few things I need to learn about setting up the ground work of Python Projects.

In testing this branch I see there are a bunch of import warnings as you say, however half of them shouldn't be a warning.
The warning has:
```
Simyan\issue\__init__.py:1:1: F401 '.issue.Issue' imported but unused
Simyan\issue\__init__.py:1:1: F401 '.issue.IssueSchema' imported but unused
Simyan\issue\__init__.py:1:1: F401 '.issue.IssueList' imported but unused
```
However it is imported by the session.py
https://github.com/Buried-In-Code/Simyan/blob/c13f9caf455ac185d78e143cb6b1d4722caabc5e/Simyan/session.py#L14

So some adjustments would need to be done.
Any suggestions? Removing them as per Flake8 breaks the session imports.

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+max-line-length = 88
```
Comment (Buried-In-Code) [inline .flake8:2]: As per the .editorconfig it should be set to 180
```suggestion
max-line-length = 180
```

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+max-complexity = 18
+select = B,C,E,F,W,T4,B9
+ignore = E501
```
Comment (Buried-In-Code) [inline .flake8:5]: I don't think it's necessary to ignore E501 as I wouldn't want to have any lines extend the max-line-length.

**Review by Buried-In-Code — CHANGES_REQUESTED** [review]

File: .flake8 (hunk 1 above)
Comment (Buried-In-Code) [reply .flake8:2]: This may be reduced from 180 in the future, but for now settings should match the .editorconfig

Comment (bpepple) [reply .flake8:2]: Ah, didn't see you had an editor config. 180 might be the largest max_line_length I've seen in a project. ;)

File: .flake8 (hunk 2 above)
Comment (bpepple) [reply .flake8:5]: No problem. Also, I would suggest using a formatter (black, yapf, pep8, etc) with the project to help maintain line-length and also improved readability for some of your more dense blocks (particularly in the session.py)

**Comment by bpepple** [comment]
> In testing this branch I see there are a bunch of import warnings as you say, however half of them shouldn't be a warning.
> The warning has:
> 
> ```
> Simyan\issue\__init__.py:1:1: F401 '.issue.Issue' imported but unused
> Simyan\issue\__init__.py:1:1: F401 '.issue.IssueSchema' imported but unused
> Simyan\issue\__init__.py:1:1: F401 '.issue.IssueList' imported but unused
> ```
> 
> However it is imported by the session.py
> 
> https://github.com/Buried-In-Code/Simyan/blob/c13f9caf455ac185d78e143cb6b1d4722caabc5e/Simyan/session.py#L14
> 
> So some adjustments would need to be done.
> Any suggestions? Removing them as per Flake8 breaks the session imports.

Probably need to look all the imports in the projects since I'm sure their is a circular dep problem going on here. If I get some free cycles this week, I'll look at creating a pull request.

**Review by Buried-In-Code — APPROVED** [review]

**Pull request merged**